Xamarin.Forms.Platform.UAP/CollectionView/CarouselViewRenderer.cs

```diff
@@ -124,33 +140,66 @@ protected override void UpdateItemTemplate()
 			ListViewBase.ItemTemplate = CarouselItemsViewTemplate;
 		}
 
-		protected override Task ScrollTo(ScrollToRequestEventArgs args)
+		void CollectionItemsSourceChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
 		{
-			var targetItem = FindCarouselItem(args);
+			var carouselPosition = Carousel.Position;
+			var currentItemPosition = FindCarouselItemIndex(Carousel.CurrentItem);
+			var count = (sender as IList).Count;
 
-			// TODO: jsuarezruiz Include support to animated scroll.
-			ListViewBase.ScrollIntoView(targetItem);
+			bool removingCurrentElement = currentItemPosition == -1;
+			bool removingLastElement = e.OldStartingIndex == count;
+			bool removingFirstElement = e.OldStartingIndex == 0;
+			bool removingCurrentElementButNotFirst = removingCurrentElement && removingLastElement && Carousel.Position > 0;
 
-			return Task.FromResult<object>(null);
+			if (removingCurrentElementButNotFirst)
+			{
+				carouselPosition = Carousel.Position - 1;
+			
+			}
+			else if (removingFirstElement && !removingCurrentElement)
+			{
+				carouselPosition = currentItemPosition;
+			}
+
+			if(removingCurrentElement)
+			{
+				SetCurrentItem(carouselPosition);
+				UpdatePosition(carouselPosition);
+			}
+			
 		}
 
 		void OnListSizeChanged(object sender, Windows.UI.Xaml.SizeChangedEventArgs e)
 		{
 			UpdateItemsSource();
 			UpdateSnapPointsType();
 			UpdateSnapPointsAlignment();
+
+			HackFixInitialPadding();
+			UpdateInitialPosition();
+		}
+
+		void HackFixInitialPadding()
+		{
+			//There's a 12 pixel on the left that i couldn't figure yet how to remove
```

Comment:
Need extra investigation why this happens

Reply:
Is it always 12 pixels no matter what the device screen size and pixel density are?

Reply:
Yeah i m not sure, and this doesn't really work tbh, I added a test for #9771 that sets the ItemsSource in a async fashion and gives me wrong results, I will keep looking better at this tomorrow. If you have any idea about this I will appreciate any help. Is like it seems there's a extra empty item in there

Reply:
FIXED! was the default padding on a ListViewItem style